### PR TITLE
Fix f-strings, parenthesized expression whitespaces, and inf/nan

### PR DESCRIPTION
**Pull request by thatbirdguythatuknownot**
…constants.

Fix the f-string output to follow PEP 9001.
Fix parenthesized expressions by adding whitespace to the beginning and end of the expression.
Fix `inf`/`nan` constants not appearing correctly.

### CODE REVIEW

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:192]: The comment, and the code implementation, contradict each other here.

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:183]: Can you give this a return type?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:189]: This can also get annotations - it's simple enough to do so.

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:215]: I like the comments here but this one seems poorly worded

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:213]: How is the comment true if `escaped_string[-1]` is `'`?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:217]: Does this ever fail? Why is it here?

Ia it just because "this should be True unless there's a bug..."?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:228]: Should we really have space here inside of the string?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:220]: There's only one empty line here

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:241]:
```suggestion
        raise ValueError(f"Unexpected node inside ast.JoinedStr: {node!r}")
```

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:350]: Didn't you also add `space()` inside `parenthesize`?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:579]: Can we add a newline here to differentiate the two code paths?

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:715]: With the second line in mind, won't this give us duplicates trailing space?

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:192]: No, they don't after close inspection. This code was just copied from `ast._Unparser`.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:215]: Going to fix it if I can offer an alternative.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:213]: I didn't write that code.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:217]: Code was just copied from `ast._Unparser` to fix the whitespace in f-strings.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:228]: > Should we really have space here inside of the string?

PEP 9001 does it.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:220]: Going to fix.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:350]: Both spaces matter. Example, when `((1 + 2) * 3)` is done.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:715]: Gonna fix.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:116]: @Bluenix2 What do you mean by "if someone wants it to be `inf`"?

Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py]: It's been removed in favor of using the source instead.

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:829]: Can you move this line to be right below `global source`? It is a bit confusing having the `ast.parse()` call between them.

File: bot/utils/blurple_formatter.py
Comment (Bluenix2) [inline bot/utils/blurple_formatter.py:827]: The formatter may be run in a thread, I am not sure if this is thread-safe.

**Comment by Bluenix2** [comment]
What do we want to do with this PR (asking others in the events team and core developers)? The event is over already.

I believe the plan is to migrate this to sir-lancebot, so do we merge this first and then move forward with that?

**Comment by minalike** [comment]
I didn't announce the end of the event until April 4, so I think this PR is fair game to merge.

For other pending issues for blurplify perhaps they should be transferred to sir-lancebot and then stalled until the feature is ported.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:583]: @Bluenix2 `quote_types` can't be modified properly if a list comprehension was used.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:574]: I'll change that.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:708]: Last `space()` in the f-string is already implemented below that line.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:829]: Doing it.

File: bot/utils/blurple_formatter.py
Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:827]: This will require a major revision to be thread-safe. Any ideas for it?

Comment (Bluenix2) [reply bot/utils/blurple_formatter.py:827]: You can add a lock (and a comment) inside of the command that calls this, that would be more light-weight than a threading lock.

Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:827]: I'm not sure I know how to do that.

Comment (thatbirdguythatuknownot) [reply bot/utils/blurple_formatter.py:827]: Is everything all good now?

**Review by Bluenix2 — APPROVED** [review]
I'll be honest in that I don't think I fully understand all 3 things this fixes, and the fact that they are in one PR and not 3 doesn't help, but I think this looks good!

Thank you for this ❤️

I think we should consider squash-merging this because of 2e0f128ed05e7905ac6d8fb6968a850b5ecb8860

**Comment by ChrisLovering** [comment]
once linting is sorted we can merge this PR

**Pull request merged**
